modules/common/route/route.go

```diff
@@ -83,8 +84,9 @@ func (r *Route) CreateOrPatch(
 ) (ctrl.Result, error) {
 	route := &routev1.Route{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      r.route.Name,
-			Namespace: r.route.Namespace,
+			Name:        r.route.Name,
+			Namespace:   r.route.Namespace,
+			Annotations: r.route.Annotations,
```

Comment:
Below we directly override existing annotations on the route. I'm wondering if we want to do a merge similarly how the lables are added.
```
		route.Labels = util.MergeStringMaps(route.Labels, r.route.Labels)
		route.Annotations = r.route.Annotations
```